### ISSUE TITLE
[1.3 beta] Regression: no users list

### ISSUE THREAD

**Issue by pdepaepe** [Graylog2/graylog2-server#1608]
Hello,

Currently testing 1.3 beta, i'm not able to get users list page.
On log side, i see:

2015-12-04T15:35:13.134Z ERROR [RestResource] Unable to load role names for role IDs [563c874ee4b001c07c5f02a9, 563c87fae4b001c07c5f0364] for user UserImpl{fields={_id=5640d260e4b02855eda908f6, username=pdepaepe, email=pdepaepe@localhost.com, permissions=[ "users:edit:pdepaepe"], password=xxxxxxxxxx, full_name=pdepaepe's account, roles=[ { "$oid" : "563c874ee4b001c07c5f02a9"} , { "$oid" : "563c87fae4b001c07c5f0364"}]}, id=5640d260e4b02855eda908f6}

When i call GET roles, i don't see 563c874ee4b001c07c5f02a9 and 563c87fae4b001c07c5f0364, so i guess, theses references are stuck in user object ?

PS: roll-backed to 1.2, users are back again

**Comment by kroepke** [comment]
Those IDs should be resolved to proper role names, so the roles call won't show them.

Still the user loading should not fail.

I'll investigate.

**Comment by bernd** [comment]
There is a regression that the role in the user object does not get deleted when a role is removed via the REST API. This will be fixed.

I can reproduce it with the following steps:

1. Create new role "Test"
1. Assign "Test" role to user
1. Remove "Reader" role from user (only works via API request)
1. Delete "Test" role via API
1. Open "System/Users" to reproduce the error

**Comment by bernd** [comment]
The commands to reproduce the problem.

1. Create role "Test"
1. Create user "john" with "Test" as only role
1. Delete role "Test"
1. Check "System/Users"

#### Commands

```
$ curl -sni -XPOST -d @role-create.json -H Content-Type:application/json localhost:12900/roles
HTTP/1.1 201 Created
Location: http://127.0.0.1:12900/roles/Test
Content-Type: application/json
Transfer-Encoding: chunked

{"name":"Test","description":"A test role","permissions":["*"],"read_only":false}

$ curl -sni -XPOST -d @user-create.json -H Content-Type:application/json localhost:12900/users
HTTP/1.1 201 Created
Location: http://127.0.0.1:12900/users/john
Transfer-Encoding: chunked

$ curl -sni -XDELETE localhost:12900/roles/Test
HTTP/1.1 204 No Content
Transfer-Encoding: chunked
```

#### role-create.json

```json
{
  "name": "Test",
  "description": "A test role",
  "permissions": ["*"]
}
```

#### user-create.json
```json
{
  "username": "john",
  "email": "john@example.com",
  "full_name": "John Doe",
  "password": "123456",
  "permissions": [
    "users:edit:john",
    "users:passwordchange:john",
    "indexercluster:read",
    "messagecount:read",
    "journal:read",
    "inputs:read",
    "metrics:read",
    "savedsearches:edit",
    "fieldnames:read",
    "buffers:read",
    "system:read",
    "savedsearches:create",
    "jvmstats:read",
    "throughput:read",
    "savedsearches:read",
    "messages:read"
  ],
  "timezone": "Europe/Berlin",
  "session_timeout_ms": -1,
  "startpage": {},
  "roles": [
    "Test"
  ]
}
```

**Comment by bernd** [comment]
This will be fixed in the upcoming Graylog 1.3 release. Thank you for the report!
